dropwizard-jetty/src/test/java/io/dropwizard/jetty/NetUtilTest.java

```diff
@@ -0,0 +1,81 @@
+package io.dropwizard.jetty;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+public class NetUtilTest {
+
+    private static final String OS_NAME_PROPERTY = "os.name";
+    public String osName;
+
+    @Before
+    public void setup() {
+        osName = System.getProperty(OS_NAME_PROPERTY);
+    }
+
+    @After
+    public void tearDown() {
+        if (osName == null) {
+            System.clearProperty(OS_NAME_PROPERTY);
+        } else {
+            System.setProperty(OS_NAME_PROPERTY, osName);
+        }
+    }
+
+    @Test
+    public void testDefaultTcpBacklogForWindows() {
+        System.setProperty(OS_NAME_PROPERTY, "win");
+        if (NetUtil.isWindows() && !isTcpBacklogSettingReadable()) {
+            assertEquals(NetUtil.DEFAULT_TCP_BACKLOG_WINDOWS, NetUtil.getTcpBacklog());
+        }
+    }
+
+    @Test
+    public void testNonWindowsDefaultTcpBacklog() {
+        System.setProperty(OS_NAME_PROPERTY, "lin");
+
+        if (!NetUtil.isWindows() && !isTcpBacklogSettingReadable()) {
+            assertEquals(NetUtil.DEFAULT_TCP_BACKLOG_LINUX, NetUtil.getTcpBacklog());
+        }
+    }
+
+    @Test
+    public void testNonWindowsSpecifiedTcpBacklog() {
+        System.setProperty(OS_NAME_PROPERTY, "lin");
+
+        if (!NetUtil.isWindows() && !isTcpBacklogSettingReadable()) {
+            assertEquals(100, NetUtil.getTcpBacklog(100));
+        }
+    }
+
+    @Test
+    public void testOsSetting() {
+        if(!NetUtil.isWindows() && isTcpBacklogSettingReadable()) {
+            assertNotEquals(-1, NetUtil.getTcpBacklog(-1));
+        }
+    }
+
+
+    public boolean isTcpBacklogSettingReadable() {
```

Comment:
You can use JUnit assumptions together with this method.

https://github.com/junit-team/junit4/wiki/Assumptions-with-assume